reference.md

```diff
@@ -4,4 +4,93 @@ layout: reference
+  * `dim() `        # check dimension of data frame
+  * `nrow() `       # returns the number of rows
+  * `ncol() `       # returns the number of  columns
+  * `tail() `       # shows the last 6 rows
+  * `names() `      # returns the column names (synonym of colnames() for data frame objects)
+  * `rownames() `   # returns the row names
+  * `summary() `    # summary statistics for each column
+  * `glimpse`       # like `str()` applied to a data frame but tries to show as much data as possible
+  * `factor() `      # create factors
+  * `levels() `      # check levels of a factor
+  * `nlevels() `     # check number of levels of a factor
+  * `as.character()` # convert an object to a character vector
+  * `as.numeric()`   # convert an object to a numeric vector
+  * `as.numeric(as.character(x))` # convert factors where the levels appear as characters to a numeric vector
+  * `as.numeric(levels(x))[x]` # convert factors where the levels appear as numbers  to a numeric vector
+  * `plot()`  # plot an object
+  * `addNA()` # convert NA into a factor level
+  * `data.frame()`  # create a data.frame object
+  * `ymd()` # convert a vector representing year, month, and day to a Date vector
+  * `paste()` # concatenate vectors after converting to character
+
+### Lesson 3 -- Data Wrangling with dplyr and tidyr
+
+  * `str()` # check structure of the object and information about the class, length and content of each column
+  * `view()` # invoke a spreadsheet-style data viewer
+  * `select() `          # select columns of a data frame
+  * `filter() `          # allows you to select a subset of rows in a data frame
+  * `%>% `               # pipes to select and filter at the same time
+  * `mutate() `          # create new columns based on the values in existing columns
+  * `head() `       # shows the first 6 rows
+  * `group_by() `        # split the data into groups, apply some analysis to each group, and then combine the results.
+  * `summarize() `       # collapses each group into a single-row summary of that group
+  * `mean()` # calculate the mean value of a vector  
+  * `!is.na()`   # test if there are no missing values
+  * `print()` # print values to the console
+  * `min()` # return the minimum value of a vector
+  * `arrange()` # arrange rows by variables
+  * `desc()` # transform a vector into a format that will be sorted in descending order
+  * `count()` # counts the total number of records for each category
+  * `pivot_wider()` # reshape a data frame by a key-value pair across multiple columns
+  * `pivot_longer()` # reshape a data frame by collapsing into a key-value pair
+  * `replace_na()`  # Replace NAs with specified values
+  * `n_distinct()` # get a count of unique values
+  * `write_csv()` # save to a csv formatted file
+
+### Lesson 4 -- Data Visualization with ggplot2
+
+  * `read_csv()` # load a csv formatted file into R memory
+  * `ggplot2(data= , aes(x= , y= )) + geom_point( ) + facet_wrap () + theme_bw() + theme() `
```

Comment:
Should we add an explanation that this is skeleton code?

Reply:
Yes, I can do that. Should it be on the same line or in a footnote?

Reply:
Hm, the rest are in the same line right after the code, so perhaps the same as the other line?

Reply:
Gotcha, didn't realize there was no description. Thanks!